enhancements/installer/aws-custom-region-and-endpoints.md

```diff
@@ -0,0 +1,402 @@
+
+Currently the installer only allows users to specify the regions that has RHCOS published AMIs, and therefore to support custom regions, the installer will allow users to specify any region string as long as the users also provide the custom endpoints for some predetermined list of services that are necessary for successful installs.
+
+To support booting machines in custom regions, the installer will always copying AMIs from `us-east-1` to the region where the user is installing, encrypting the AMI using the user's default account KMS key. This is has known limitations for increased AMI copy times and also increased cost of transfer across regions. If the target region doesn't support copying AMI from `us-east-1` region, the users are expected to provide the source AMI in the install-config using the platform [configuration][install-config-aws-ami]. The installer will use the AMI provided and copy-encrypt to an AMI for the cluster, keeping the behavior consistent with other workflows.
+
+### User Stories
+
+#### US1: Using private VPC endpoints
+
+#### US2: Special endpoints like FIPS endpoints for US Gov Cloud
+
+#### US3: Installing to new public regions without native support for RHCOS or AWS SDK
+
+#### US4: Installing to custom region
+
+## Implementation Details/Notes/Constraints
+
+### Installer
+
+#### Install Config
+
+The user can provide a list of API endpoints for various services.
+
+```go
+// ServiceEndpoint store the configuration of a url to
+// override existing defaults of AWS Services.
+type ServiceEndpoint struct {
+    Service string `json:"service"`
+
+    // This must be a HTTPS URL
+    URL     string `json:"url"`
+}
+```
+
+```go
+type Platform struct {
+    ...
+    // ServiceEndpoints list contains custom endpoints which will override default
+    // service endpoint of AWS Services.
+    // There must be only one ServiceEndpoint for a service.
+    // +optional
+    ServiceEndpoints []ServiceEndpoint `json:"serviceEndpoints,omitempty"`
+    ...
+}
+```
+
+The yaml representation by the user would look like:
+
+```yaml
+platform:
+  aws:
+    serviceEndpoints:
+    - service: ec2
+      url: https://ec2.custom.url
+    - service: s3
+      url: https://s3.custom.url
+```
+
+#### Validations for service endpoints
+
+1. The installer must ensure that only one override for service is specified by the user.
+
+2. The installer should ensure that the service endpoint is at least reachable from the host.
+
+3. The URL for the service endpoint must be `https` and the host should trust the certificate.
+
+#### Configuring the AWS SDK
```

Comment:
cc @alexander-demichev  for aws actuator

Reply:
@abhinavdahiya For QE's testing, do we need some extra configuration on installer client for this section? Or this section is mainly for developer?